packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx

```diff
@@ -81,13 +112,20 @@ const PersonalInformationStep = ({
               }
               name="username"
               hookToForm
+              onInput={(e) => setCurrentUsername(e.target.value.trim())}
+              customError={isUsernameTaken ? 'Username already exists' : ''}
             />
             <CWButton
               label="Generate random username"
               buttonType="tertiary"
               buttonHeight="sm"
               type="button"
               containerClassName="random-generate-btn"
+              onClick={() => {
+                const randomUsername = generateUsername('', 2);
+                setValue('username', randomUsername);
+                setCurrentUsername(randomUsername);
+              }}
```

Comment:
can you move it to the handler function like `handleGenerateUsername`

Reply:
hm, nice suggestion. Actually, i am working on passing refs for formMethods to CWForm in https://github.com/hicommonwealth/commonwealth/issues/6750, so i think it would be good if we cleanup there.